tezos/context/src/serialize/mod.rs

```diff
@@ -306,11 +312,17 @@ pub fn deserialize_hash_id(data: &[u8]) -> Result<(Option<HashId>, usize), Deser
     }
 }
 
-pub fn serialize_hash_id(
-    hash_id: u64,
+pub fn serialize_hash_id_impl(
+    hash_id: Option<HashId>,
     output: &mut Vec<u8>,
+    repository: &mut ContextKeyValueStore,
     stats: &mut SerializeStats,
 ) -> Result<(), SerializationError> {
+    let hash_id = match hash_id {
+        Some(hash_id) => repository.make_hash_id_ready_for_commit(hash_id)?.as_u64(),
+        None => 0,
```

Comment:
@sebastiencs how does this work? I am not clear on this logic here with a possible 0 and also the next line that gets the max hash id.

Reply:
A `HashId` serialized to `0` will be deserialized as `Option::<HashId>::None`.
We have to handle this case for the parent of a commit, sometimes it can have no parent:
https://github.com/tezedge/tezedge/blob/develop/tezos/context/src/serialize/in_memory.rs#L538

The max hash id is only for statistics, it was useful for the in-memory context, to get an idea of the highest `HashId`